spinalcordtoolbox/image.py

```diff
@@ -404,24 +404,22 @@ def set_qform_to_sform(self):
         self.hdr.set_qform(self.hdr.get_sform())
         self.hdr._structarr['qform_code'] = self.hdr._structarr['sform_code']
 
-    def fix_header_dtype(self, header='self', array='self'):
+    def fix_header_dtype(self):
         """
         Change the header dtype to the match the datatype of the array.
-
-        Note: This method defaults to operating on the image's `data` and `hdr` properties,
-              but the fix can also be applied to supplied arrays and headers, too."""
-        hdr = self.hdr if header == 'self' else header
-        data = self.data if array == 'self' else array
-        dtype_header = hdr.get_data_dtype()
-        dtype_data = data.dtype
-        if dtype_data != dtype_header:
-            if dtype_data == 'bool':
-                # Sometimes we create boolean arrays, apparently! But, using 'bool' for nibabel headers triggers:
-                # `nibabel.spatialimages.HeaderDataError: data dtype "bool" not supported`, so use `uint8` instead
-                dtype_data = np.uint8
+        """
+        # Using bool for nibabel headers is unsupported, so use uint8 instead:
+        # `nibabel.spatialimages.HeaderDataError: data dtype "bool" not supported`
+        dtype_data = self.data.dtype
+        if dtype_data == bool:
+            dtype_data = np.uint8
+
+        dtype_header = self.hdr.get_data_dtype()
+        if dtype_header != dtype_data:
             logger.warning(f"Image header specifies datatype '{dtype_header}', but array is of type "
                            f"'{dtype_data}'. Header metadata will be overwritten to use '{dtype_data}'.")
-            hdr.set_data_dtype(dtype_data)
+
+        self.hdr.set_data_dtype(dtype_data)
```

Comment:
This is a tiny tiny tiny nitpicky question, but I noticed the call to `self.hdr.set_data_dtype` was moved outside of the `if` statement.

Is there any reason to call `self.hdr.set_data_dtype()` unconditionally, even when the two datatypes are the same?

Reply:
There's no particular reason. I think how it happened is that:
1. I first rewrote the code without any reference to `dtype_header`, so that `set_data_dtype` was called unconditionally.
2. Then I noticed the conditional `logger.warning` in the old code and inserted it back.

Either way should be fine. Feel free to hoist it back into the conditional. After all, if you wondered why it was this way, that's proof that it's confusing to the reader.